tls/s2n_connection.c

```diff
@@ -1554,3 +1554,16 @@ int s2n_connection_get_config(struct s2n_connection *conn, struct s2n_config **c
 
     return S2N_SUCCESS;
 }
+
+/*
+ * Set the client_hello callback behavior to polling.
+ *
+ * Polling means that the callback function can be called multiple times.
+ */
+int s2n_connection_client_hello_cb_enable_poll(struct s2n_connection *conn) {
```

Comment:
Should this really be a connection-level setting? Wouldn't this belong on the config, where the callback itself is set? I know we talked about this offline, but I don't remember the conclusion.

Reply:
Why does the callback need to set the flag? Wouldn't it be true for all instances of a callback? If it only needs to be polled once based on logic in the callback itself, wouldn't you just call the done method then?

Reply:
I should clarify. For the Rust bindings, the application callback wont actually be responsible for making this call. Instead the rust bindings will handle the setup. **When using rust bindings, the poll functionality will always be enabled.**

That being said, this is an internal API so its a one way door and we can change how this is done in the future if we want.

Reply:
I don't see the motivation for putting it on the client hello structure though. What is the benefit that justifies manually setting it on every connection?

Reply:
One nice thing is that it avoids putting the fields on the connection/config objects which is used across the entire codebase.

Reply:
spoke offline:
- move the flag to the config

why:
- The flag pertains to the client_hello callback, which lives on the config, so the flag should also live on the config.
- Setting the flag on the config will make it easy to use for other use cases outside of the Rust bindings also.

Reply:
resolved in https://github.com/aws/s2n-tls/pull/3230/commits/2858e16f2c47dc25b6cbf1fdabca30440ac8704b

Reply:
This should be moved to s2n_config and should take a config directly. The name should also be updated.
```suggestion
 */
int s2n_config_client_hello_cb_enable_poll(struct s2n_config *config) {
```